aws/ecs_cluster.go

```diff
@@ -0,0 +1,105 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// tagging an ECS cluster with a given tag key-value pair and a cluster ARN
+func tagEcsCluster(awsSession *session.Session, clusterArn *string, tagKey string, tagValue string) (*ecs.TagResourceOutput, error) {
+	svc := ecs.New(awsSession)
+	input := &ecs.TagResourceInput{
+		ResourceArn: clusterArn,
+		Tags: []*ecs.Tag{
+			{
+				Key:   aws.String(tagKey),
+				Value: aws.String(tagValue),
+			},
+		},
+	}
+
+	result, err := svc.TagResource(input)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	return result, nil
+}
+
+func getAllEcsClustersOlderThan(awsSession *session.Session, region string, t time.Time) ([]*string, error) {
+	// get all ecs clusters
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+
+	clusterArns, err := getAllEcsClusters(awsSession)
+	if err != nil {
+		logging.Logger.Errorf("Error getting all ECS clusters")
+		return nil, err
+	}
+
+	// var filteredEcsClusters []*string
+	for _, clusterArn := range clusterArns {
+		// if firstSeenTag is present
+		// check if cluster is older than t
+		// then include in filteredEcsClusters
+		//	else
+		// set first seen tag to time now
+
+		firstSeen := getClusterTag(awsSession, clusterArn, "first_seen")
+
+		if firstSeen == nil {
+			tagEcsCluster(awsSession, clusterArn, "first_seen", "tomorrow")
```

Comment:
Always handle the `error` returned by a method. Otherwise, it get silently swallowed, which will make debugging very tough. Note, most IDEs support warning you about this.

```suggestion
			_, err := tagEcsCluster(awsSession, clusterArn, "first_seen", "tomorrow")
			if err != nil {
			    return nil, err
			}
```